graspologic/plot/plot.py

```diff
@@ -673,6 +676,223 @@ def pairplot(
+        msg = "if give labels array must give palette dicts or string desciptors"
+        raise ValueError(msg)
+    elif labels is None and cluster_palette is None and label_palette is None:
+        colors = sns.color_palette("Set1", num_comps)
+        cluster_palette = dict(zip(np.unique(lab_names), colors))
+        label_palette = dict(zip(np.unique(lab_names), colors))
+    elif labels is None and (cluster_palette is not None or label_palette is not None):
+        msg = "must give labels if using palettes for labels and clusters"
+        raise ValueError(msg)
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        if X.shape[1] == 2:
+            pairplot, axes = plt.subplots(1, 1, figsize=(12, 12))
+            _plot_ellipse(
+                data,
+                X,
+                0,
+                1,
+                means,
+                covariances,
+                axes,
+                label_palette,
+                cluster_palette,
+                alpha=alpha,
+            )
+            axes.set_ylabel("Dim " + str(0))
+            axes.set_xlabel("Dim " + str(1))
+            axes.label_outer()
+            axes.spines["right"].set_visible(False)
+            axes.spines["top"].set_visible(False)
+            if title:
+                pairplot.suptitle(title)
+            else:
+                pairplot.suptitle(
+                    "Fit a Gaussian mixture using "
+                    + str(means.shape[0])
+                    + " components"
+                )
+            return
+        pairplot, axes = plt.subplots(X.shape[1], X.shape[1], figsize=(12, 12))
+        for k in range(X.shape[1]):
+            for j in range(X.shape[1]):
+                if k == j:
+                    for t, lab in zip([i for i in range(X.shape[1])], label_palette):
+                        sns.distplot(
+                            X[Y_ == t, k],
+                            kde=True,
+                            ax=axes[k, k],
+                            color=label_palette[lab],
+                        )
+                    axes[k, j].set_xticks([]), axes[k, j].set_yticks([])
+                    axes[k, j].set_ylabel(None)
+                    axes[k, j].set_ylabel(j), axes[k, j].set_xlabel(k)
+                else:
+                    _plot_ellipse(
+                        data,
+                        X,
+                        j,
+                        k,
+                        means,
+                        covariances,
+                        axes[k, j],
+                        label_palette,
+                        cluster_palette,
+                        alpha=alpha,
+                    )
+        for i in range(X.shape[1]):
+            for j in range(X.shape[1]):
+                axes[i, j].set_ylabel("Dim " + str(i))
+                axes[i, j].set_xlabel("Dim " + str(j))
+        if title:
+            pairplot.suptitle(title)
+        else:
```

Comment:
all this stuff is nested really deep, and there are a lot of lines of code. I feel like there has to be a way to simplify/flatten out this code a bit, and make more concise or break into more pieces?

Reply:
I really am unsure about this. The main reason it occupies so many lines of code is because of the formatting criterion. Beyond that, I feel the looping is necessary considering I am building a pairplot from scratch without using any other premade graphs. It is difficult to read, but I am unsure how to make it cleaner

Reply:
I believe if we use the graspologic pairplot instead of starting from scratch we will avoid this nesting. So i will resolve this and then i will address this in a new PR.